src/bindings/c/ov/src/ov_c_api.cpp

```diff
@@ -0,0 +1,1428 @@
+
+ov_element_type_e find_ov_element_type_e(ov::element::Type type) {
+    for (auto iter = element_type_map.begin(); iter != element_type_map.end(); iter++) {
+        if (iter->second == type) {
+            return iter->first;
+        }
+    }
+    return ov_element_type_e::UNDEFINED;
+}
+
+#define GET_OV_ELEMENT_TYPE(a) element_type_map[a]
+#define GET_CAPI_ELEMENT_TYPE(a) find_ov_element_type_e(a)
+
+#define CATCH_OV_EXCEPTION(StatusCode, ExceptionType) catch (const InferenceEngine::ExceptionType&) {return ov_status_e::StatusCode;}
+
+#define CATCH_OV_EXCEPTIONS                                         \
+        CATCH_OV_EXCEPTION(GENERAL_ERROR, GeneralError)             \
+        CATCH_OV_EXCEPTION(NOT_IMPLEMENTED, NotImplemented)         \
+        CATCH_OV_EXCEPTION(NETWORK_NOT_LOADED, NetworkNotLoaded)    \
+        CATCH_OV_EXCEPTION(PARAMETER_MISMATCH, ParameterMismatch)   \
+        CATCH_OV_EXCEPTION(NOT_FOUND, NotFound)                     \
+        CATCH_OV_EXCEPTION(OUT_OF_BOUNDS, OutOfBounds)              \
+        CATCH_OV_EXCEPTION(UNEXPECTED, Unexpected)                  \
+        CATCH_OV_EXCEPTION(REQUEST_BUSY, RequestBusy)               \
+        CATCH_OV_EXCEPTION(RESULT_NOT_READY, ResultNotReady)        \
+        CATCH_OV_EXCEPTION(NOT_ALLOCATED, NotAllocated)             \
+        CATCH_OV_EXCEPTION(INFER_NOT_STARTED, InferNotStarted)      \
+        CATCH_OV_EXCEPTION(NETWORK_NOT_READ, NetworkNotRead)        \
+        CATCH_OV_EXCEPTION(INFER_CANCELLED, InferCancelled)         \
+        catch (...) {return ov_status_e::UNEXPECTED;}
+
+char* str_to_char_array(const std::string& str) {
+    char *char_array = new char[str.length() + 1];
+    std::copy_n(str.begin(), str.length() + 1, char_array);
+    return char_array;
+}
+
+const char* ov_get_error_info(ov_status_e status) {
+    if (status > ov_status_e::UNKNOWN_ERROR)
+        return error_infos[ov_status_e::UNKNOWN_ERROR];
+    return error_infos[status];
+}
+
+ov_status_e ov_get_version(ov_version_t *version) {
+    if (!version) {
+        return ov_status_e::GENERAL_ERROR;
+    }
+
+    try {
+        ov::Version object = ov::get_openvino_version();
+
+        std::string version_builderNumber = object.buildNumber;
+        version->buildNumber = str_to_char_array(version_builderNumber);
+
+        std::string version_description = object.description;
+        version->description = str_to_char_array(version_description);
+    } CATCH_OV_EXCEPTIONS
+    return ov_status_e::OK;
+}
+
+void ov_version_free(ov_version_t *version) {
+    if (!version) {
+        return;
+    }
+    delete[] version->buildNumber;
+    version->buildNumber = nullptr;
+    delete[] version->description;
+    version->description = nullptr;
+}
+
+ov_status_e ov_core_create(const char *xml_config_file, ov_core_t **core) {
+    if (!core || !xml_config_file) {
+        return ov_status_e::GENERAL_ERROR;
+    }
+
+    try {
+        *core = new ov_core_t;
```

Comment:
I suppose we need to use unique_ptr like in old API to prevent memory leaks in case of exception on next lines